### PR TITLE
Restore Edit option for connections

### PR DESCRIPTION
**Pull request by antony-britto**
# TITLE

## Description
Allow edit for connections, reverting https://github.com/cdapio/cdap-ui/pull/179 and https://github.com/cdapio/cdap-ui/pull/122

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement

## Links
Jira: [1004](https://cdap.atlassian.net/browse/PLUGIN-1004)

### CODE REVIEW

File: app/cdap/components/Connections/Browser/SidePanel/CategorizedConnections.tsx [hunk 1]
```diff
@@ -203,7 +203,7 @@ export function CategorizedConnections({
   const [initConnConfig, setInitConnConfig] = React.useState(null);
   const [connectionToDelete, setConnectionToDelete] = React.useState(null);
   const [deleteError, setDeleteError] = React.useState(null);
-  const [isView, setIsView] = React.useState(false);
+  const [isEdit, setIsEdit] = React.useState(false);
```
Comment (seanfeld) [inline app/cdap/components/Connections/Browser/SidePanel/CategorizedConnections.tsx]: nit: import {useState} from react

File: app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx [hunk 2]
```diff
@@ -60,35 +60,33 @@ export function ConnectionConfigForm({
...
-          value={name}
-          onChange={(v) => {
-            if (!isView) {
-              setName(v.name);
-            }
-          }}
-          disabled={isView}
-          extraConfig={{ properties: {} }}
-        />
+        <If condition={!isEdit}>
```
Comment (seanfeld) [inline app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx]: don't use the If component anymore

File: app/cdap/components/Connections/Create/index.tsx [hunk 3]
```diff
@@ -237,7 +235,7 @@ export function CreateConnection({
 
   return (
     <div className={classes.root}>
-      <If condition={!hadInitial && state.activeStep === ICreateConnectionSteps.CONNECTOR_CONFIG}>
+      <If condition={state.activeStep === ICreateConnectionSteps.CONNECTOR_CONFIG}>
```
Comment (seanfeld) [inline app/cdap/components/Connections/Create/index.tsx]: I know this is existing code but i think we should update things when we edit them (remove if component)

**Review by seanfeld — APPROVED** [review]
Code looks fine - just nits if you want to address them.

File: app/cdap/components/Connections/Browser/SidePanel/CategorizedConnections.tsx (hunk 1 above)
Comment (antony-britto) [reply app/cdap/components/Connections/Browser/SidePanel/CategorizedConnections.tsx]: Done

File: app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx (hunk 2 above)
Comment (antony-britto) [reply app/cdap/components/Connections/Create/ConnectionConfiguration/ConnectionConfigForm.tsx]: Done

File: app/cdap/components/Connections/Create/index.tsx (hunk 3 above)
Comment (antony-britto) [reply app/cdap/components/Connections/Create/index.tsx]: Done

**Review by njbriggs — APPROVED** [review]

**Pull request merged**